packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -249,6 +249,41 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  /// Track the visbility of the start of the selected text.
+  ///
+  /// For example, if the text contains "Hello World", and the user selects
+  /// "Hello", then scrolls so only "World" is visible, this will become
+  /// 'false'. If the user scrolls back so that the "H" is visible again, this
+  /// will become 'true'.
+  final ValueNotifier<bool> selectionStartInView = ValueNotifier<bool>(true);
```

Comment:
you should expose a `ValueListenable<bool>` and have the `ValueNotifier` be private. Otherwise other code can affect the value.

Reply:
Done.